code/modules/mob/living/carbon/slime/slime_extractinator.dm

```diff
@@ -0,0 +1,115 @@
+/obj/machinery/slime_extractinator
+	name = "slime core extractinator"
+	desc = "A bulky machine that, when fed a slime corpse, rapidly extracts the held cores."
+	desc_info = "This machine can be upgraded with a micro laser to increase its extraction speed, or a matter bin to increase its slime capacity. It will place slime extracts into a slime extract bag if it's adjacent to the machine."
+	icon = 'icons/obj/stationobjs.dmi'
+	icon_state = "slime_extractor"
+	density = TRUE
+	anchored = TRUE
+	use_power = TRUE
+	var/slime_limit = 2 // how many slimes we can process at a time
+	var/extraction_speed = 150
+	idle_power_usage = 15
+	active_power_usage = 50
+	var/list/mob/living/carbon/slime/extract_slimes = list()
+
+	component_types = list(
+		/obj/item/circuitboard/slime_extractinator,
+		/obj/item/stock_parts/matter_bin,
+		/obj/item/stock_parts/micro_laser,
+		/obj/item/stack/cable_coil{amount = 5}
+	)
+
+/obj/machinery/slime_extractinator/examine(mob/user)
+	. = ..()
+	to_chat(user, FONT_SMALL(SPAN_NOTICE("It can hold <b>[slime_limit] slime\s</b> at a time.")))
+	if(length(extract_slimes))
+		to_chat(user, FONT_SMALL(SPAN_WARNING("It is currently processing <b>[length(extract_slimes)] slime\s</b>.")))
+
+/obj/machinery/slime_extractinator/update_icon()
+	cut_overlays()
+	if(panel_open)
+		var/mutable_appearance/panel_overlay = mutable_appearance(icon, "[icon_state]-panel")
+		add_overlay(panel_overlay)
+	if(length(extract_slimes))
+		var/mutable_appearance/interior_overlay = mutable_appearance(icon, "[icon_state]-interior", EFFECTS_ABOVE_LIGHTING_LAYER)
+		add_overlay(interior_overlay)
+		var/mutable_appearance/spinning_overlay = mutable_appearance(icon, "[icon_state]-running", EFFECTS_ABOVE_LIGHTING_LAYER)
+		add_overlay(spinning_overlay)
+		set_light(2.5, 1, COLOR_VIOLET)
+	else
+		set_light(FALSE)
+
+/obj/machinery/slime_extractinator/RefreshParts()
+	..()
+	slime_limit = initial(slime_limit)
+	extraction_speed = initial(extraction_speed)
+
+	for(var/obj/item/stock_parts/P in component_parts)
+		if(ismatterbin(P))
+			slime_limit *= P.rating
+		else if(ismicrolaser(P))
+			extraction_speed = extraction_speed / P.rating
+
+/obj/machinery/slime_extractinator/attackby(obj/item/O, mob/user)
+	if(length(extract_slimes))
+		to_chat(user, SPAN_WARNING("You can't modify \the [src] while it's busy. Please wait for the completion of previous operation."))
+		return
+
+	if(default_deconstruction_screwdriver(user, O))
+		return
+	if(default_deconstruction_crowbar(user, O))
+		return
+	if(default_part_replacement(user, O))
+		return
```

Comment:
what's this for?

Reply:
those are used to open the panel, deconstruct with a crowbar, or upgrade using an RPED